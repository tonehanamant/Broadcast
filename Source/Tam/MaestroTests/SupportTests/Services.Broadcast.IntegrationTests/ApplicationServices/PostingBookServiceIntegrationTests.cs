using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using IntegrationTests.Common;
using NUnit.Framework;
using Services.Broadcast.ApplicationServices;

namespace Services.Broadcast.IntegrationTests.ApplicationServices
{
    public class PostingBookServiceIntegrationTests
    {
        private readonly IPostingBookService _PostingBookService = IntegrationTestApplicationServiceFactory.GetApplicationService<IPostingBookService>();

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetShareBooks()
        {
            var shareBooks = _PostingBookService.GetShareBooks(new DateTime(2019,12,31));

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(shareBooks));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetHUTBooksByFlightStartDate()
        {
            var shareBooks = _PostingBookService.GetHUTBooks(new DateTime(2019, 12, 31), null);

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(shareBooks));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetHUTBooksByShareBookId()
        {
            var shareBooks = _PostingBookService.GetHUTBooks(null, 437);

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(shareBooks));
        }
    }
}
