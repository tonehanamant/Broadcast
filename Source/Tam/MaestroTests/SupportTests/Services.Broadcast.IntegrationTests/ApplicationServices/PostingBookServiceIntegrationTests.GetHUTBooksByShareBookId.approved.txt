[
  {
    "Display": "Feb 2017",
    "Id": 425
  },
  {
    "Display": "Nov 2016",
    "Id": 422
  },
  {
    "Display": "Jul 2016",
    "Id": 418
  },
  {
    "Display": "May 2016",
    "Id": 416
  },
  {
    "Display": "Feb 2016",
    "Id": 413
  },
  {
    "Display": "Jul 2015",
    "Id": 406
  },
  {
    "Display": "May 2015",
    "Id": 404
  },
  {
    "Display": "Feb 2015",
    "Id": 401
  }
]